frontend/src/utils/api.js

```diff
@@ -45,9 +45,12 @@ const api = {
   datasets: {
     get: id => get(`/datasets/${id}/`),
     list: (options, abort) => get("/datasets/", options, { abort }),
-    setReleaseStatus: (id, release_status, exceptions = [], filters = {}) => patch(`/datasets/${id}/set_release_status/`, { release_status, exceptions, filters })
   },
-
+  readsets: {
+    get: id => get(`/readsets/${id}`),
+    list: (options, abort) => get(`/readsets/`, options, { abort }),
+    setReleaseStatus: readset => post(`/readsets/set_release_status/`, readset),
```

Comment:
You should add options to your actions to be able to get either the regular readset or the readsetWithMetrics...